### PR TITLE
feat: support certificate chain

### PR DESCRIPTION
**Pull request by chloeyin**
1. Update `notation-go` and `notation-core-go`
2. Support certificate chain

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,8 +5,8 @@ go 1.17
 require (
 	github.com/Azure/azure-sdk-for-go v65.0.0+incompatible
 	github.com/Azure/go-autorest/autorest v0.11.24
-	github.com/notaryproject/notation-core-go v0.1.0-alpha.3
-	github.com/notaryproject/notation-go v0.10.0-alpha.3.0.20220819145229-a9fb055b8d92
+	github.com/notaryproject/notation-core-go v0.1.0-alpha.3.0.20220929020818-b5df54c7465b
```
Comment (shizhMSFT) [inline go.mod:8]: `notation-core-go v0.1.0-alpha.4` has been released. Let's update to that.

File: internal/cloud/keyvault.go [hunk 2]
```diff
@@ -42,8 +42,8 @@ const (
 	authorizerFromCLI clientAuthMethod = "AKV_AUTH_FROM_CLI"
 
 	// defaultAuthMethod is the default auth method if user doesn't provide an environment variable
-	// the default value will be authorizerFromMI
-	defaultAuthMethod = authorizerFromMI
+	// the default value will be authorizerFromCLI
+	defaultAuthMethod = authorizerFromCLI
```
Comment (shizhMSFT) [inline internal/cloud/keyvault.go:46]: @yizha1 @FeynmanZhou Please confirm the default behavior is expected.

File: internal/crypto/cert.go [hunk 3]
```diff
@@ -28,6 +22,40 @@ func ParseCertificates(data []byte) ([]*x509.Certificate, error) {
...
+		// skip private key
+		if block.Type == "CERTIFICATE" {
+			cert, err := x509.ParseCertificate(block.Bytes)
+			if err != nil {
+				return nil, err
+			}
+			certs = append(certs, cert)
+		}
+	}
+	return certs, err
```
Comment (shizhMSFT) [inline internal/crypto/cert.go:50]:
```suggestion
	return certs, nil
```

File: internal/cloud/keyvault.go (hunk 2 above)
Comment (yizha1) [reply internal/cloud/keyvault.go:46]: Confirm the default is CLI, which is also recommended by @dtzar .

File: go.mod (hunk 1 above)
Comment (yizha1) [reply go.mod:8]: Maybe we should wait for notation-go alpha.4 release and update the dependency accordingly.

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: go.mod (hunk 1 above)
Comment (shizhMSFT) [reply go.mod]: We can bump it in later PRs.

**Pull request merged**